lib/DDG/Rewrite.pm

```diff
@@ -156,8 +156,11 @@ sub _build_nginx_conf {
 
 	$cfg .= "\techo_before_body '".$self->callback."(';\n" if $wrap_jsonp_callback;
 	$cfg .= "\techo_before_body '".$self->callback.qq|("';\n| if $wrap_string_callback;
+	my ($spice_name) = $self->path =~ /js\/spice\/(.+?)\//;
+	my $upstream = '$'.$spice_name.'_upstream';
+	$cfg .= "\tset ". $upstream . ' ' . $scheme."://".$host.":".$port.";\n";
```

Comment:
I gather that `set` creates a variable that we use below?  Just one set of double quote around the entire thing for clarity?